hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -195,122 +198,174 @@ data Environment = Environment
   , otherParties :: [Party]
   }
 
--- | The heart of the Hydra head logic, a handler of all kinds of 'Event' in the
--- Hydra head. This may also be split into multiple handlers, i.e. one for hydra
--- network events, one for client events and one for main chain events, or by
--- sub-'State'.
-update ::
+-- | Simplified Hydra Head Protocol Without Conflict Resolution
+onIdleHandleClientInitEvent :: Party -> [Party] -> ContestationPeriod -> Outcome tx
+onIdleHandleClientInitEvent party otherParties contestationPeriod =
+  NewState IdleState [OnChainEffect (InitTx parameters)]
+ where
+  parameters =
+    HeadParameters
+      { contestationPeriod
+      , parties = party : otherParties
+      }
+
+onIdleHandleChainInitTx :: [Party] -> ContestationPeriod -> Outcome tx
+onIdleHandleChainInitTx parties contestationPeriod =
+  NewState
+    ( InitialState
+        { parameters = HeadParameters{contestationPeriod, parties}
+        , pendingCommits = Set.fromList parties
+        , committed = mempty
+        , previousRecoverableState = IdleState
+        }
+    )
+    [ClientEffect $ ReadyToCommit $ fromList parties]
+
+onInitialHandleClientCommitEvent ::
+  HeadState tx -> Event tx -> Party -> PendingCommits -> UTxOType tx -> Outcome tx
+onInitialHandleClientCommitEvent _ _ party pendingCommits utxo
+  | canCommit = OnlyEffects [OnChainEffect (CommitTx party utxo)]
+ where
+  canCommit = party `Set.member` pendingCommits
+onInitialHandleClientCommitEvent st ev _ _ _ = Error $ InvalidEvent ev st
```

Comment:
I think is a valid idea, but imo is better to return an invalid event instead, as it carries information.
Unhandled looks potentially hard to debug.
Wdyt?

Reply:
For this case, we actually introduced a bug, that has been already fixed, but instead of InvalidEvent we should of have been returning CommandFailed as in the base case.
Not sure if the suggestion still applies now.
Wdyt?